scripts/ci/style/pylint_check.sh

```diff
@@ -4,17 +4,27 @@ name=$(basename "$0")
 
 source ./scripts/ci/helpers.sh
 
-echo "Running pylint..."
+GIT_ROOT=$(git rev-parse --show-toplevel)
 
 set_on_failed_callback "FAIL pylint errors"
 
-pylint --rcfile="./pyproject.toml" bentoml | tee /tmp/"$name"_bentoml
-pylint --rcfile="./pyproject.toml" --disable=E0401,F0010 tests docker | tee /tmp/"$name"_tests_docker
+if [[ -n "$GITHUB_BASE_REF" ]]; then
+  echo "Running pylint on changed files..."
+  git fetch origin "$GITHUB_BASE_REF"
+  if ! (git diff --name-only --diff-filter=d "origin/$GITHUB_BASE_REF" -z -- '*.py' | xargs -0 --no-run-if-empty pylint --rcfile="$GIT_ROOT/pyproject.toml" --exit-zero); then
+    FAIL "pylint failed."
+    exit 1
+  fi
+else
+  echo "Running pylint for the whole library..."
```

Comment:
also I think pylint is failing on main branch now,  for the same reason pyright was failing before

Reply:
We can just disable pylint for running on the main branch then